lua/entities/gmod_wire_expression2/init.lua

```diff
@@ -356,8 +363,19 @@ function ENT:ResetContext()
 	end
 
 	for k, v in pairs(self.Inputs) do
-		if wire_expression_types[v.Type][3] then
-			self.GlobalScope[k] = wire_expression_types[v.Type][3](self.context, v.Value)
+		local CopyFunction = wire_expression_types[v.Type][3]
+		if CopyFunction then
+			local Copy = CopyFunction(self.context, v.Value)
```

Comment:
This is still unnecessarily calling `CopyFunction` when `self.rawinputs[k]` exists.